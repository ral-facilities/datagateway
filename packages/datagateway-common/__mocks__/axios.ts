import axios from 'axios';

const requests = {
  get: vi.fn(() => Promise.resolve({ data: {} })),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
  isAxiosError: axios.isAxiosError,
};

export default requests;
