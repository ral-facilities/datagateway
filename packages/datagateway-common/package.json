{
  "name": "datagateway-common",
  "version": "0.1.0",
  "private": true,
  "files": [
    "lib"
  ],
  "main": "./lib/index.js",
  "dependencies": {
    "@date-io/date-fns": "^1.3.13",
    "@material-ui/pickers": "^3.2.10",
    "@material-ui/lab": "^4.0.0-alpha.57",
    "@types/lodash.debounce": "^4.0.6",
    "axios": "^0.21.1",
    "clsx": "^1.1.1",
    "connected-react-router": "^6.9.1",
    "date-fns": "^2.17.0",
    "i18next": "^19.9.0",
    "lodash.debounce": "^4.0.8",
    "loglevel": "^1.7.1",
    "react-draggable": "^4.4.3",
    "react-i18next": "^11.8.8",
    "react-redux": "^7.1.0",
    "react-router": "^5.0.1",
    "react-scripts": "4.0.3",
    "react-virtualized": "^9.22.3",
    "redux": "^4.0.4",
    "redux-mock-store": "^1.5.4",
    "redux-thunk": "^2.3.0"
  },
  "peerDependencies": {
    "@material-ui/core": ">=  4.0.0 < 5",
    "@material-ui/icons": ">=  4.0.0 < 5",
    "react": ">= 16.8.0 < 17",
    "react-dom": ">= 16.8.0 < 17",
    "react-router-dom": ">= 5.0.0 < 6"
  },
  "devDependencies": {
    "@material-ui/core": "^4.11.3",
    "@material-ui/icons": "^4.11.2",
    "@types/enzyme": "^3.10.8",
    "@types/jest": "^26.0.4",
    "@types/node": "^14.14.31",
    "@types/react": "^17.0.2",
    "@types/react-router-dom": "^5.1.7",
    "@types/react-virtualized": "^9.21.10",
    "@typescript-eslint/eslint-plugin": "^4.16.1",
    "@typescript-eslint/parser": "^4.16.1",
    "babel-eslint": "10.1.0",
    "enzyme": "^3.10.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "enzyme-to-json": "^3.6.1",
    "eslint": "^7.5.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-config-react-app": "^6.0.0",
    "eslint-plugin-cypress": "^2.11.1",
    "eslint-plugin-flowtype": "^5.8.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.20.3",
    "eslint-plugin-react-hooks": "^4.1.0",
    "lint-staged": "^10.5.4",
    "prettier": "^2.2.1",
    "react": "^16.13.1",
    "react-dom": "^16.11.0",
    "react-router-dom": "^5.0.1",
    "typescript": "4.2.2",
    "tslib": "^2.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom --coverage --watchAll=false",
    "prepare": "yarn tsc",
    "test:watch": "react-scripts test --env=jsdom --watch",
    "lint:js": "eslint --ext=tsx --ext=ts --ext=js --ext=jsx --fix ./src",
    "tsc": "tsc -p tsconfig.lib.json",
    "eject": "react-scripts eject",
    "pre-commit": "lint-staged"
  },
  "lint-staged": {
    "src/**/*.{tsx,ts,js,jsx,json}": [
      "eslint --max-warnings=0 --ext=tsx --ext=ts --ext=js --ext=jsx  --fix",
      "prettier --write"
    ],
    "cypress/**/*.{tsx,ts,js,jsx,json}": [
      "eslint --max-warnings=0 --ext=tsx --ext=ts --ext=js --ext=jsx --fix",
      "prettier --write"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ]
  },
  "jest": {
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "collectCoverageFrom": [
      "src/**/*.{tsx,ts,js,jsx}",
      "!src/index.tsx",
      "!src/serviceWorker.ts",
      "!src/setupTests.js"
    ]
  }
}
