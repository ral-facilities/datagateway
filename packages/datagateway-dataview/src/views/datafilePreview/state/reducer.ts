import DATAFILE_PREVIEWER_DEFAULT from '../defaults';
import {
  DecrementDatafilePreviewerZoomLevelType,
  IncrementDatafilePreviewerZoomLevelType,
  ResetDatafilePreviewerZoomLevelType,
  ToggleDatafilePreviewerDetailsPanePayload,
  ToggleDatafilePreviewerDetailsPaneType,
} from './actions';
import { DGDataViewState } from '../../../state/app.types';

/**
 * Defines the shape of the redux state for {@link DatafilePreviewer}
 */
interface DatafilePreviewerState {
  /**
   * Whether the details pane should be shown or not.
   */
  isDetailsPaneShown: boolean;

  /**
   * The current zoom level of the previewer, in percentage. For example,
   * 100% maps to 100, 110% maps to 110, 80% maps to 80, etc.
   */
  zoomLevel: number;
}

function toggleDatafilePreviewerDetailsPane(
  state: DGDataViewState,
  payload: ToggleDatafilePreviewerDetailsPanePayload
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      isDetailsPaneShown: payload.shouldShow,
    },
  };
}

function incDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      zoomLevel:
        state.datafilePreviewer.zoomLevel +
        DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep,
    },
  };
}

function decDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      // zoom level should not go lower than zoomLevelStep
      // otherwise will result in zoom level <= 0.
      zoomLevel: Math.max(
        DATAFILE_PREVIEWER_DEFAULT.minZoomLevel,
        state.datafilePreviewer.zoomLevel -
          DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep
      ),
    },
  };
}

function resetDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    datafilePreviewer: {
      ...state.datafilePreviewer,
      zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
    },
  };
}

const datafilePreviewerInitialState: DatafilePreviewerState = {
  isDetailsPaneShown: DATAFILE_PREVIEWER_DEFAULT.showDetailsPane,
  zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
};

const datafilePreviewerReducer = {
  [ToggleDatafilePreviewerDetailsPaneType]: toggleDatafilePreviewerDetailsPane,
  [IncrementDatafilePreviewerZoomLevelType]: incDatafilePreviewerZoomLevel,
  [DecrementDatafilePreviewerZoomLevelType]: decDatafilePreviewerZoomLevel,
  [ResetDatafilePreviewerZoomLevelType]: resetDatafilePreviewerZoomLevel,
};

export {
  datafilePreviewerInitialState,
  datafilePreviewerReducer,
  toggleDatafilePreviewerDetailsPane,
  incDatafilePreviewerZoomLevel,
  decDatafilePreviewerZoomLevel,
  resetDatafilePreviewerZoomLevel,
};
export type { DatafilePreviewerState };
