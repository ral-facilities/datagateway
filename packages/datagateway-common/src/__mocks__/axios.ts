// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn(() => Promise.resolve({ data: {} })),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  AxiosError: axios.AxiosError,
};

export default requests;
