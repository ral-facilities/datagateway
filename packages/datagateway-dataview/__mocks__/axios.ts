import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  delete: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
