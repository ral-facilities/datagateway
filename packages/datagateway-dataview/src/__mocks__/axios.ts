// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path === '/datagateway-dataview-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'Generic',
          idsUrl: 'ids',
          apiUrl: 'api',
          downloadApiUrl: 'download',
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
  isAxiosError: axios.isAxiosError,
};

export default requests;
