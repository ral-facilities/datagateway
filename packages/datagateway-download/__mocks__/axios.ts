import axios from 'axios';

const requests = {
  get: vi.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: vi.fn(() => Promise.resolve({ data: {} })),
  put: vi.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
