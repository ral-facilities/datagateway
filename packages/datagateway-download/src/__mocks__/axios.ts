// TODO: move __mocks__ folder back to package root once facebook/create-react-app#7539 is fixed

import axios from 'axios';

const requests = {
  get: jest.fn((path) => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'https://example.com/api',
          downloadApiUrl: 'https://example.com/downloadApi',
          idsUrl: 'https://example.com/ids',
          fileCountMax: 5000,
          totalSizeMax: 1000000000000,
          accessMethods: {
            https: {
              idsUrl: 'https://example.com/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://example.com/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
          routes: [
            {
              section: 'Test',
              link: '/download',
              displayName: 'Download',
              order: 0,
            },
          ],
        },
      });
    } else if (path.includes('/topcat/user/downloads')) {
      return Promise.resolve({ data: [] });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  put: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};

export default requests;
