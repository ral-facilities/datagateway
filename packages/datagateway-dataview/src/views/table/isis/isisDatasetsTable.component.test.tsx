import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import ISISDatasetsTable from './isisDatasetsTable.component';
import { initialState as dgDataViewInitialState } from '../../../state/reducers/dgdataview.reducer';
import configureStore from 'redux-mock-store';
import { StateType } from '../../../state/app.types';
import {
  fetchDatasetsRequest,
  filterTable,
  sortTable,
  fetchDatasetDetailsRequest,
  downloadDatasetRequest,
  fetchDatasetCountRequest,
  addToCartRequest,
  removeFromCartRequest,
  dGCommonInitialState,
  fetchAllIdsRequest,
} from 'datagateway-common';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Table } from 'datagateway-common';
import { MemoryRouter } from 'react-router';
import axios from 'axios';
import { push } from 'connected-react-router';

describe('ISIS Dataset table component', () => {
  let shallow;
  let mount;
  let mockStore;
  let state: StateType;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'ISISDatasetsTable' });
    mount = createMount();

    mockStore = configureStore([thunk]);
    state = JSON.parse(
      JSON.stringify({
        dgdataview: dgDataViewInitialState,
        dgcommon: dGCommonInitialState,
      })
    );
    state.dgcommon.data = [
      {
        id: 1,
        name: 'Test 1',
        size: 1,
        modTime: '2019-07-23',
        createTime: '2019-07-23',
      },
    ];
    state.dgcommon.allIds = [1];

    (axios.get as jest.Mock).mockImplementation(() =>
      Promise.resolve({ data: [] })
    );
    (axios.post as jest.Mock).mockImplementation(() =>
      Promise.resolve({ data: {} })
    );
    (axios.delete as jest.Mock).mockImplementation(() =>
      Promise.resolve({ data: {} })
    );
    global.Date.now = jest.fn(() => 1);
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <ISISDatasetsTable
        store={mockStore(state)}
        studyHierarchy={false}
        instrumentId="1"
        instrumentChildId="2"
        investigationId="3"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('sends fetchDatasetCount and fetchDatasets actions when watched store values change', () => {
    let testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    // simulate clearTable action
    testStore = mockStore({
      ...state,
      dgdataview: { ...state.dgdataview, sort: {}, filters: {} },
    });
    wrapper.setProps({ store: testStore });

    expect(testStore.getActions()[0]).toEqual(fetchDatasetCountRequest(1));
    expect(testStore.getActions()[1]).toEqual(fetchAllIdsRequest(1));
    expect(testStore.getActions()[2]).toEqual(fetchDatasetsRequest(1));
  });

  it('sends fetchDatasets action when loadMoreRows is called', () => {
    const testStore = mockStore(state);
    const wrapper = shallow(
      <ISISDatasetsTable
        studyHierarchy={false}
        instrumentId="1"
        instrumentChildId="2"
        investigationId="3"
        store={testStore}
      />
    );

    wrapper.prop('loadMoreRows')({ startIndex: 50, stopIndex: 74 });

    expect(testStore.getActions()[0]).toEqual(fetchDatasetsRequest(1));
  });

  it('sends filterTable action on text filter', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    const filterInput = wrapper
      .find('[aria-label="Filter by datasets.name"] input')
      .first();
    filterInput.instance().value = 'test';
    filterInput.simulate('change');

    expect(testStore.getActions()[3]).toEqual(
      filterTable('name', { value: 'test', type: 'include' })
    );

    filterInput.instance().value = '';
    filterInput.simulate('change');

    expect(testStore.getActions()[5]).toEqual(filterTable('name', null));
  });

  it('sends filterTable action on date filter', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    const filterInput = wrapper.find(
      '[aria-label="datasets.modified_time date filter to"]'
    );
    filterInput.instance().value = '2019-08-06';
    filterInput.simulate('change');

    expect(testStore.getActions()[3]).toEqual(
      filterTable('modTime', { endDate: '2019-08-06' })
    );

    filterInput.instance().value = '';
    filterInput.simulate('change');

    expect(testStore.getActions()[5]).toEqual(filterTable('modTime', null));
  });

  it('sends sortTable action on sort', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    wrapper
      .find('[role="columnheader"] span[role="button"]')
      .first()
      .simulate('click');

    expect(testStore.getActions()[3]).toEqual(sortTable('name', 'asc'));
  });

  it('sends addToCart action on unchecked checkbox click', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    wrapper.find('[aria-label="select row 0"]').first().simulate('click');

    expect(testStore.getActions()[3]).toEqual(addToCartRequest());
  });

  it('sends removeFromCart action on checked checkbox click', () => {
    state.dgcommon.cartItems = [
      {
        entityId: 1,
        entityType: 'dataset',
        id: 1,
        name: 'test',
        parentEntities: [],
      },
    ];

    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    wrapper.find('[aria-label="select row 0"]').first().simulate('click');

    expect(testStore.getActions()[3]).toEqual(removeFromCartRequest());
  });

  it('selected rows only considers relevant cart items', () => {
    state.dgcommon.cartItems = [
      {
        entityId: 1,
        entityType: 'investigation',
        id: 1,
        name: 'test',
        parentEntities: [],
      },
      {
        entityId: 2,
        entityType: 'dataset',
        id: 2,
        name: 'test',
        parentEntities: [],
      },
    ];

    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    const selectAllCheckbox = wrapper
      .find('[aria-label="select all rows"]')
      .first();

    expect(selectAllCheckbox.prop('checked')).toEqual(false);
    expect(selectAllCheckbox.prop('data-indeterminate')).toEqual(false);
  });

  it('renders details panel correctly and it sends actions', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            store={testStore}
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    const detailsPanelWrapper = shallow(
      wrapper.find(Table).prop('detailsPanel')({
        rowData: state.dgcommon.data[0],
      })
    );
    expect(detailsPanelWrapper).toMatchSnapshot();

    mount(
      wrapper.find(Table).prop('detailsPanel')({
        rowData: state.dgcommon.data[0],
        detailsPanelResize: jest.fn(),
      })
    );

    expect(testStore.getActions()[3]).toEqual(fetchDatasetDetailsRequest());

    detailsPanelWrapper.find('#dataset-datafiles-tab').simulate('click');
    expect(testStore.getActions()).toHaveLength(5);
    expect(testStore.getActions()[4]).toEqual(
      push(
        '/browse/instrument/1/facilityCycle/2/investigation/3/dataset/1/datafile'
      )
    );
  });

  it('renders dataset name as a link', () => {
    const wrapper = mount(
      <Provider store={mockStore(state)}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    expect(
      wrapper.find('[aria-colindex=3]').find('p').children()
    ).toMatchSnapshot();
  });

  it('renders dataset name as a link in StudyHierarchy', () => {
    const wrapper = mount(
      <Provider store={mockStore(state)}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={true}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    expect(
      wrapper.find('[aria-colindex=3]').find('p').children()
    ).toMatchSnapshot();
  });

  it('sends downloadData action on click of download button', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISDatasetsTable
            studyHierarchy={false}
            instrumentId="1"
            instrumentChildId="2"
            investigationId="3"
          />
        </MemoryRouter>
      </Provider>
    );

    wrapper.find('button[aria-label="datasets.download"]').simulate('click');

    expect(testStore.getActions()[3]).toEqual(downloadDatasetRequest(1));
  });
});
