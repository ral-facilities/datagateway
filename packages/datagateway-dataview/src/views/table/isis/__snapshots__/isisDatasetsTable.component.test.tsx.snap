// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISIS Dataset table component renders correctly 1`] = `
Object {
  "actions": Array [
    [Function],
  ],
  "allIds": Array [
    1,
  ],
  "columns": Array [
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "name",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "datasets.name",
    },
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "size",
      "disableSort": true,
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "datasets.size",
    },
    Object {
      "dataKey": "createTime",
      "defaultSort": "desc",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "datasets.create_time",
    },
    Object {
      "dataKey": "modTime",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "datasets.modified_time",
    },
  ],
  "data": Array [
    Object {
      "createTime": "2019-07-23",
      "id": 1,
      "modTime": "2019-07-23",
      "name": "Test 1",
      "size": 1,
    },
  ],
  "detailsPanel": [Function],
  "disableSelectAll": false,
  "loadMoreRows": [Function],
  "loading": false,
  "onCheck": [MockFunction],
  "onSort": [Function],
  "onUncheck": [MockFunction],
  "selectedRows": Array [],
  "sort": Object {
    "createTime": "desc",
  },
  "totalRowCount": 0,
}
`;

exports[`ISIS Dataset table component renders dataset name as a link 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
  href="/browse/instrument/1/facilityCycle/2/investigation/3/dataset/1"
>
  Test 1
</a>
`;

exports[`ISIS Dataset table component renders dataset name as a link in StudyHierarchy 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
  href="/browseStudyHierarchy/instrument/1/study/2/investigation/3/dataset/1"
>
  Test 1
</a>
`;
